src/libtriton/CMakeLists.txt

```diff
@@ -272,14 +272,17 @@ target_include_directories(triton BEFORE PUBLIC
 
 # Link Triton's dependencies
 target_link_libraries(triton PUBLIC
-    ${PYTHON_LIBRARIES}
     ${Boost_LIBRARIES}
     ${Z3_LIBRARIES}
     ${LLVM_LIBRARIES}
     ${BITWUZLA_LIBRARIES}
     ${CAPSTONE_LIBRARIES}
 )
 
+if(PYTHON_BINDINGS)
+    target_link_libraries(triton PRIVATE Python3::Module)
```

Comment:
According to unit tests, i'm not sure it should be private.

Reply:
Yeah I’ll fix this next week. I think there is an issue with the way the unit tests are set up. The triton shared library (pyd python module) should just import from python. The unit tests shouldn’t have to know about python at all?

Reply:
Maybe to clarify a bit: a PUBLIC dependency would become a part of the Triton CMake package. This seems incorrect conceptually, because a package consumer does not need to know whether the python bindings are included in the shared library or not (and thus doesn’t have to find Python, etc)

Reply:
On my side, I had to add `target_link_libraries(triton PRIVATE ${Python3_LIBRARIES})` in additon to the existing one inside the `PYTHON_BINDINGS` if block:
```cmake
if(PYTHON_BINDINGS)
    target_link_libraries(triton PRIVATE ${Python3_LIBRARIES})
    target_link_libraries(triton PRIVATE Python3::Module)
endif()
```
Otherwise I got `undefine reference` errors when it tries to link the python library. Any idea why @mrexodia ?

Reply:
In theory this is not necessary, but welcome to CMake/Python land where nothing makes sense 🤷‍♂️

Reply:
Ok, I'll add that line, rebase and merge then. Everything else seems to be working fine.